src/NuGet.Clients/NuGet.SolutionRestoreManager/VSNominationUtilities.cs

```diff
@@ -408,18 +408,21 @@ private static LibraryDependency ToPackageLibraryDependency(IVsReferenceItem ite
             return dependency;
         }
 
-        private static DownloadDependency ToPackageDownloadDependency(IVsReferenceItem item)
+        private static IEnumerable<DownloadDependency> ToPackageDownloadDependency(IVsReferenceItem item)
         {
             var id = item.Name;
-            var versionRange = GetVersionRange(item);
-            if (!(versionRange.HasLowerAndUpperBounds && versionRange.MinVersion.Equals(versionRange.MaxVersion)))
+            var versionRanges = GetVersionRangeList(item);
+            foreach (var versionRange in versionRanges)
```

Comment:
To avoid confusion, I either:

a) Use LINQ to make the string.Split operation and apply `ToPackageDownloadDependency` to each result.
b) Remove LINQ code and use only foreach loops.

Combining foreach loops with LINQ may lead to confusion.